openrouteservice/src/main/java/org/heigit/ors/centrality/algorithms/brandes/BrandesCentralityAlgorithm.java

```diff
@@ -0,0 +1,148 @@
+package org.heigit.ors.centrality.algorithms.brandes;
+
+import com.graphhopper.routing.weighting.Weighting;
+import com.graphhopper.storage.Graph;
+import com.graphhopper.storage.NodeAccess;
+import com.graphhopper.util.EdgeExplorer;
+import com.graphhopper.util.EdgeIterator;
+import com.vividsolutions.jts.geom.Coordinate;
+import org.heigit.ors.centrality.CentralityResult;
+import org.heigit.ors.centrality.algorithms.CentralityAlgorithm;
+
+import java.io.FileWriter;
+import java.io.PrintStream;
+import java.util.*;
+
+public class BrandesCentralityAlgorithm implements CentralityAlgorithm {
+    protected Graph graph;
+    protected Weighting weighting;
+
+    public void init(Graph graph, Weighting weighting)
+    {
+        this.graph = graph;
+        this.weighting = weighting;
+    }
+
+    private class QueueElement implements Comparable<QueueElement> {
+        public Double dist;
+        public Integer pred;
+        public Integer v;
+
+        public QueueElement(Double dist, Integer pred, Integer v) {
+            this.dist = dist;
+            this.pred = pred;
+            this.v = v;
+        }
+
+        public int compareTo(QueueElement other) {
+            return Double.compare(this.dist, other.dist);
+        }
+    }
+    // this implementation follows the code given in
+    // "A Faster Algorithm for Betweenness Centrality" by Ulrik Brandes, 2001
+    public HashMap<Integer, Double> compute(ArrayList<Integer> nodesInBBox) throws Exception {
+        HashMap<Integer, Double> betweenness = new HashMap<>();
+
+        // c_b[v] = 0 forall v in V
+        for (int v: nodesInBBox) {
+            betweenness.put(v,0.0d);
+        }
+
+
+        for (int s : nodesInBBox) {
+            Stack<Integer> S = new Stack<>();
+            HashMap<Integer, List<Integer>> P = new HashMap<>();
+            HashMap<Integer, Integer> sigma = new HashMap<>();
+
+            // single source shortest path
+            //S, P, sigma = SingleSourceDijkstra(graph, nodesInBBox, s);
```

Comment:
This is just a question, nothing to be done: could one of the existing routing algorithms be used here?

Reply:
Not without any modification - as far as I know, no routing algorithm keeps track of predecessors on shortest paths or even considers alternative shortest paths of same length.